fw/src/mppt/global/firefly.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file firefly.hpp
+ * @author Praneel Murali (praneelm@utexas.edu)
+ * @brief firefly MPPT algorithm driver.
+ * @version 0.2.0
+ * @date 2024-01-20
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+
+#pragma once
+#include "../mppt.hpp"
+#include <vector>
+#include <random>
+#include <algorithm>
+
+class FireflyAlgorithm : public MPPT {
+public:
+    FireflyAlgorithm(unsigned int populationSize = 25, float gamma = 1.0f, float beta0 = 2.0f, float alpha = 0.3f)
```

Comment:
highly recommend documentation here explaining each parameter and how someone might determine how set their own

Reply:
and for each new method introduced not inherited from mppt.hpp, really